urllib3/util/retry.py

```diff
@@ -290,5 +289,17 @@ def __repr__(self):
                     cls=type(self), self=self)
 
 
+def redirect_method(method, status):
```

Comment:
I vote for the Retry object just because it's easiest to replace/override/pass in.